### DISCUSSION TITLE
Casting from nb::ndarray<T> to nb::ndarray<>?

### DISCUSSION THREAD

**Discussion by mreineck** [wjakob/nanobind#532] [Q&A]
I'm experimenting with porting a fairly large numerical library (https://gitlab.mpcdf.mpg.de/mtr/ducc) from `pybind11` to `nanobind`. Things aren't looking too bad, but one recurring theme is that I cannot implicitly cast an array of a well-defined type (i.e. `nb::ndarray<T>`) to a generic `nb::ndarray<>`. In `pybind11` there was an implicit conversion from `py::array_t<T>` to `py::array`. Is there a way for me to emulate this?

**Comment by wjakob** [comment]
Does this not work? It is part of the `ndarray` class definition.

```
    template <typename... Args2>
    explicit ndarray(const ndarray<Args2...> &other) : ndarray(other.m_handle) { }
```

The idea here is that you hopefully know what you are doing. Nanobind performs no checks.

**Reply by mreineck** [reply]
Thanks for pointing out that the constructor is explicit, that was the bit I was missing!
I'll check this as soon as possible.

Concerning the missing checks ... I still try to get a feeling for nanobind and its relation to pybind11, so it is very possible that I'm doing stupid things. But I'm not in a hurry with the migration and I'm rather confident to get everything sorted out in time ;-)
